### ISSUE TITLE
Endless loading of the editor

### ISSUE THREAD

**Issue by xperia-droid** [navaddins/TinyMCEEditor-CAL#1]
Hi Navaddins Team,

we tried to integrate your Editor in our Dynamics NAV BC 14 CU 10 (DE-Version) System. But it shows only an endless Loading Screen.
![image](https://user-images.githubusercontent.com/24463981/106746569-0513bc00-6623-11eb-9f60-78cc71eb80e5.png)

We have already made countless attempts to get the editor displayed.

We have set up a completely new VM, with a CRONUS DEMO database. In this VM, we have imported the Fob objects accordingly and inserted the Control Add-In. In the VM, the editor works completely without any problems.

During the VM setup with the editor, the VM was offline all the time and the editor works without any problems. We thought in the VM it works without any problem, let's export the database, RTC files and NST files from the VM and integrate them in the existing infrastructure. No chance when calling the RTC, we see a completely endless loading screen again.

We don't have any group policies active in our infrastructure that could hinder this. We have also completely uninstalled the antivirus software, restarted the server and tested again.

A colleague had already included some "Alerts("line number")" in the TinyMCEEditorHelper.js file, these were also reported back, but no display of the editor.

We have also already switched to online mode, but this did not change the result at all.

So far we used the NAV.Pad from Waldo: https://github.com/waldo1001/Waldo.NAV.
But he uses the Tiny MCE Editor in version 4, which is only supported until 25.02. There we have already asked for an update to v5, because we have not managed to update it ourselves.

Therefore we also hope for you, if you have an idea, why we get an endless loading screen.

Best Regards,
Daniel

**Comment by navaddins** [comment]
@NAV-Management ,
Thank for your feedback and very sorry for my code. I try to simulate on BC 14 - Build 46080 but sad to say, I don't get any error like what you mention. Could you help me to test at your side.

Could you put alert message under "script/TinyMCEEditorHelper.js"
![image](https://user-images.githubusercontent.com/33715070/106989819-f1e12780-67ad-11eb-98ef-bb32a22e7c39.png)

and some changes on Page 90001.

![image](https://user-images.githubusercontent.com/33715070/106989960-42f11b80-67ae-11eb-80ab-a309fa115a60.png)

Do you have a chance to test with web client as well? Could you please check the dll is blocked as well.

Regards,

**Comment by xperia-droid** [comment]
@navaddins ,

thank you very much for your reply.

I have made the changes in the js file and in the page.
![image](https://user-images.githubusercontent.com/24463981/107015832-80997880-679d-11eb-89a8-3d5855a38233.png)
![image](https://user-images.githubusercontent.com/24463981/107015965-a757af00-679d-11eb-9375-b980b2b4c00b.png)
![image](https://user-images.githubusercontent.com/24463981/107015809-79726a80-679d-11eb-8c1d-e4fbc95880c8.png)


On the RTC side it behaves like this after I updated the add-in resource and opened the page:
![image](https://user-images.githubusercontent.com/24463981/107010564-cacb2b80-6796-11eb-8b92-fc0b22ebc265.png)
![image](https://user-images.githubusercontent.com/24463981/107010594-d4ed2a00-6796-11eb-9f33-b9ddf4727da8.png)

The message from line 266 does not appear :(.


On the WebClient side it behaves like this:
![image](https://user-images.githubusercontent.com/24463981/107010724-02d26e80-6797-11eb-9b2d-08eec165ab1f.png)
![image](https://user-images.githubusercontent.com/24463981/107010776-154ca800-6797-11eb-92a8-1fb856524d80.png)
![image](https://user-images.githubusercontent.com/24463981/107010820-27c6e180-6797-11eb-995c-02c880d8a7ec.png)
![image](https://user-images.githubusercontent.com/24463981/107010880-34e3d080-6797-11eb-86ec-a9d4f07203f1.png)

And then it shows the editor in the WebClient. But when I enter a single character, the first message appears again:
![image](https://user-images.githubusercontent.com/24463981/107010998-5a70da00-6797-11eb-93ca-f2f10e969677.png)

It could be that one of my colleagues changed something that this message appears every time.


I created a new NAV system with the CRONUS demo company in it, added the editor fob, changed the page 90001, added the changed resource.zip file and in the web client it behaves like this:
![image](https://user-images.githubusercontent.com/24463981/107015061-ab370180-679c-11eb-80f6-00bc7319422e.png)
![image](https://user-images.githubusercontent.com/24463981/107015125-bb4ee100-679c-11eb-837e-65f3e6ae612a.png)
![image](https://user-images.githubusercontent.com/24463981/107015206-c99cfd00-679c-11eb-8fb5-20a179ba4a1e.png)
![image](https://user-images.githubusercontent.com/24463981/107015241-d1f53800-679c-11eb-8fee-12aea4fa52a4.png)

So the WebClient works with the CRONUS NAV system. But on the RTC side it is exactly the same behavior as seen in the screenshots above.

Do you have any idea why the message for line 266 is not triggered on the rtc?
The rtc is currently the most used client by my colleagues. So it would be best if we can get it to work on the rtc. 

Best regards, and thanks for reading :).

**Comment by xperia-droid** [comment]
I forgot to add the details about the dll-file:
![image](https://user-images.githubusercontent.com/24463981/107017309-3dd8a000-679f-11eb-9bc8-56753416db09.png)


I rebuilt the dll on our side again, thinking a custom build dll might help. But the custom dll does not change anything.

**Comment by xperia-droid** [comment]
Update: 

I resetted all Objects and truncated the table. Now it works at least in the web Client:
![image](https://user-images.githubusercontent.com/24463981/107020845-a590ea00-67a3-11eb-9e07-d726cd91f811.png)

But the RTC behaviour doesn't change.

**Comment by navaddins** [comment]
@NAV-Management,
again, sorry for making troublesome to you. it's really hard to debug for RTC and without seeing because it's always regenerate the add in file. Anyway, I want to try my best for you.

Could you comment the code on Page 90001
Line No. 264 and 265.

we need to narrow down the scope for check.

Regards,

**Comment by xperia-droid** [comment]
@navaddins ,

Thank you very much for at least checking against the issue.

Code Modification:
![image](https://user-images.githubusercontent.com/24463981/107028937-85b2f380-67ae-11eb-8b91-7d5d9a4064f1.png)

RTC-Test:
![RTC-Test](https://user-images.githubusercontent.com/24463981/107029518-6e283a80-67af-11eb-88af-40f3355e4625.gif)

**Comment by navaddins** [comment]
@NAV-Management,

I have spent 10 hrs on this issue and finally, I figure out the solution. Actually, issue is not come from control add-in. It's come from IE. IE is blocking to run the JavaScript.

**Solution**

_You must close all opening RTC client before change the setting._

![image](https://user-images.githubusercontent.com/33715070/107122382-74630780-68d2-11eb-8baa-0b2178da2477.png)

You can uncheck it back when you Control AddIn is working well.

**Comment by xperia-droid** [comment]
Thank you very much for the analysis, you were able to find out the problem. We have wasted at least a hundred hours to solve the problem. 

Thanks again for the research, I mean it :).

![image](https://user-images.githubusercontent.com/24463981/107123390-95f5cc00-689d-11eb-838f-ed5f3b281c67.png)


After I disabled the setting again, the editor still worked. But I think we will enforce this setting globally in our company. 

I still wish you a nice restful weekend :).
